src/pubsub.c

```diff
@@ -501,19 +493,20 @@ int pubsubPublishMessageInternal(robj *channel, robj *message, pubsubtype type)
         channel = getDecodedObject(channel);
         while((de = dictNext(di)) != NULL) {
             robj *pattern = dictGetKey(de);
-            list *clients = dictGetVal(de);
+            dict *clients = dictGetVal(de);
             if (!stringmatchlen((char*)pattern->ptr,
                                 sdslen(pattern->ptr),
                                 (char*)channel->ptr,
                                 sdslen(channel->ptr),0)) continue;
-
-            listRewind(clients,&li);
-            while ((ln = listNext(&li)) != NULL) {
-                client *c = listNodeValue(ln);
+            dictEntry *entry;
+            dictIterator *iter = dictGetSafeIterator(clients);
```

Comment:
Do we need a safe iterator?

Reply:
i think we do, since the client can get disconnected due to client buffer limits.
i.e. safe iterators are used when the dict you're iterating over can be modified from inside the iteration.
removing it means the fingerprint check can fail, but i'm not sure if we have a test that covers that.